### ISSUE TITLE
Save() fails for collection with implicit ID

### ISSUE THREAD

**Issue by ghost** [kalcok/jc#2]
```
panic: reflect: call of reflect.Value.Interface on zero Value [recovered]
	panic: reflect: call of reflect.Value.Interface on zero Value

goroutine 21 [running]:
testing.tRunner.func1(0xc42013c2d0)
	/opt/local/lib/go/src/testing/testing.go:711 +0x2d2
panic(0x11db360, 0xc4200fc5c0)
	/opt/local/lib/go/src/runtime/panic.go:491 +0x283
reflect.valueInterface(0x0, 0x0, 0x0, 0x1, 0x0, 0x0)
	/opt/local/lib/go/src/reflect/value.go:930 +0x1bf
reflect.Value.Interface(0x0, 0x0, 0x0, 0x121ba71, 0x3)
	/opt/local/lib/go/src/reflect/value.go:925 +0x44
github.com/kalcok/jc.(*Collection).Save(0xc420132120, 0xc420132101, 0x0, 0x0, 0x1082806)
	/Users/kauchman/Documents/projects/golang/jc/src/github.com/kalcok/jc/collection.go:74 +0x3c4
github.com/kalcok/jc/tests.TestInsertImplicitIDCollection(0xc42013c2d0)
	/Users/kauchman/Documents/projects/golang/jc/src/github.com/kalcok/jc/tests/collection_test.go:69 +0x9a
testing.tRunner(0xc42013c2d0, 0x12276e0)
	/opt/local/lib/go/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
	/opt/local/lib/go/src/testing/testing.go:789 +0x2de
```

**Comment by ghost** [comment]
Issue with inserting was solved in #4 . However more testing needs to be done for SELECTing and UPDATE-ing of collections with implicit ID field

**Comment by ghost** [comment]
Tests now cover SELECT/UPDATE/INSERT of structs with implicit ID
